tests/plugins/os/unix/bsd/osx/test__os.py

```diff
@@ -25,7 +25,10 @@ def test_unix_bsd_osx_os(target_osx_users, fs_osx):
 
     assert dissect_user.name == "_dissect"
     assert dissect_user.passwd == "*"
-    assert dissect_user.home == "/var/empty"
+    # When the test is running under windows, comparing to the string representation ensures
+    # that the home directory is stored in posix format.
+    # The __eq__ operator of the path class is too lenient.
+    assert str(dissect_user.home) == "/var/empty"
```

Comment:
```suggestion
    assert dissect_user.home == "/var/empty"
```

The flow.record paths were explicitly designed so we could do this in tests. No need for casts or comments.

Reply:
I think an explicit type check for posix_path is merited as that is what it should be by definition.

Reply:
I have added an assertion for `posix_path`. 
 
In  the long term, I am still slightly unsatisfied by our current solution :) 

-  I think the original assertion `assert dissect_user.home == "/var/empty"` should have caught the bug this pr fixes.
- Related, but I think adding type checks cause the test to become unnecessarily coupled to implementation details (in this case `posix_path`). Ideally, I think we should check for behavior. 

But perhaps this is only the first time we have a bug like this. I have made a note.